### PR TITLE
Add Slurm-GCP v6 version of zone policies blueprint

### PR DESCRIPTION
**Pull request by alyssa-sm**
This PR adds a Slurm-GCP v6 version of `zone-policies-slurm-v5.yaml`.

Calling `sinfo` shows the two partitions.
```
PARTITION AVAIL  TIMELIMIT  NODES  STATE NODELIST
multiz       up   infinite      4  idle~ slurmgcpv6-multizonalnode-[0-3]
zonal        up   infinite      4  idle~ slurmgcpv6-zonalnodegroup-[0-3]
```

Manually submitting the job `sbatch -N 1 -p zonal --wrap "sleep 10"` successfully completes.
```
JobId=1 JobName=wrap
   UserId=alyssasm GroupId=alyssasm MCS_label=N/A
   Priority=1 Nice=0 Account=(null) QOS=normal
   JobState=COMPLETED Reason=None Dependency=(null)
   Requeue=1 Restarts=0 BatchFlag=1 Reboot=0 ExitCode=0:0
   RunTime=00:00:10 TimeLimit=UNLIMITED TimeMin=N/A
   SubmitTime=2024-03-22T03:48:02 EligibleTime=2024-03-22T03:48:02
   AccrueTime=2024-03-22T03:48:02
   StartTime=2024-03-22T03:49:28 EndTime=2024-03-22T03:49:38 Deadline=N/A
   SuspendTime=None SecsPreSuspend=0 LastSchedEval=2024-03-22T03:48:02 Scheduler=Main
   Partition=zonal AllocNode:Sid=slurmgcpv6-controller:3288
   ReqNodeList=(null) ExcNodeList=(null)
   NodeList=slurmgcpv6-zonalnodegroup-0
   BatchHost=slurmgcpv6-zonalnodegroup-0
   NumNodes=1 NumCPUs=1 NumTasks=1 CPUs/Task=1 ReqB:S:C:T=0:0:*:*
   ReqTRES=cpu=1,mem=7552M,node=1,billing=1
   AllocTRES=cpu=1,mem=7552M,node=1,billing=1
   Socks/Node=* NtasksPerN:B:S:C=0:0:*:* CoreSpec=*
   MinCPUsNode=1 MinMemoryCPU=7552M MinTmpDiskNode=0
   Features=(null) DelayBoot=00:00:00
   OverSubscribe=NO Contiguous=0 Licenses=(null) Network=(null)
   Command=(null)
```

Manually submitting the job `sbatch -N 1 -p multiz --wrap "sleep 10"` successfully completes. Looking into the node's configuration shows that it used `us-central1-a` which is one of the additional zones.
```
JobId=2 JobName=wrap
   UserId=alyssasm GroupId=alyssasm MCS_label=N/A
   Priority=1 Nice=0 Account=(null) QOS=normal
   JobState=COMPLETED Reason=None Dependency=(null)
   Requeue=1 Restarts=0 BatchFlag=1 Reboot=0 ExitCode=0:0
   RunTime=00:00:10 TimeLimit=UNLIMITED TimeMin=N/A
   SubmitTime=2024-03-22T03:48:14 EligibleTime=2024-03-22T03:48:14
   AccrueTime=2024-03-22T03:48:14
   StartTime=2024-03-22T03:49:28 EndTime=2024-03-22T03:49:38 Deadline=N/A
   SuspendTime=None SecsPreSuspend=0 LastSchedEval=2024-03-22T03:48:14 Scheduler=Main
   Partition=multiz AllocNode:Sid=slurmgcpv6-controller:3288
   ReqNodeList=(null) ExcNodeList=(null)
   NodeList=slurmgcpv6-multizonalnode-0
   BatchHost=slurmgcpv6-multizonalnode-0
   NumNodes=1 NumCPUs=1 NumTasks=1 CPUs/Task=1 ReqB:S:C:T=0:0:*:*
   ReqTRES=cpu=1,mem=7552M,node=1,billing=1
   AllocTRES=cpu=1,mem=7552M,node=1,billing=1
   Socks/Node=* NtasksPerN:B:S:C=0:0:*:* CoreSpec=*
   MinCPUsNode=1 MinMemoryCPU=7552M MinTmpDiskNode=0
   Features=(null) DelayBoot=00:00:00
   OverSubscribe=NO Contiguous=0 Licenses=(null) Network=(null)
   Command=(null)
```

### CODE REVIEW

File: tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    settings:
+      node_count_dynamic_max: 4
+      machine_type: n2-standard-2
+      disable_public_ips: false
+      enable_placement: false
+
+  - id: zonal_partition
+    source: community/modules/compute/schedmd-slurm-gcp-v6-partition
+    use:
+    - zonal_node_group
```
Comment (harshthakkar01) [inline tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml:56]: nit" use: [zonal_node_group]

File: tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    settings:
+      node_count_dynamic_max: 4
+      machine_type: n2-standard-2
+      enable_placement: false
+      zones: $(vars.additional_zones)
+
+  - id: multizonal_partition
+    source: community/modules/compute/schedmd-slurm-gcp-v6-partition
+    use:
+    - multizonal_node_group
```
Comment (harshthakkar01) [inline tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml:73]: nit" use: [multizonal_node_group]

File: tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    settings:
+      partition_name: multiz
+
+  - id: slurm_login
+    source: community/modules/scheduler/schedmd-slurm-gcp-v6-login
+    use:
+    - network
```
Comment (harshthakkar01) [inline tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml:80]: nit --> use: [network]

File: tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+
+blueprint_name: slurm-gcp-v6-hpc-centos7
+
+vars:
+  project_id:  ## Set GCP Project ID Here ##
+  deployment_name: slurm-gcp-v6
```
Comment (harshthakkar01) [inline tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml:21]: I think we should rename this to specific to zonal-policies. slurm-gcp-v6 sounds very generic.

**Comment by harshthakkar01** [comment]
Also it would be helpful to put the results (the node is created to us-central1-a/ us-central1-c) in canonical bug.

File: tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml (hunk 1 above)
Comment (alyssa-sm) [reply tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml]: Done!

File: tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml (hunk 2 above)
Comment (alyssa-sm) [reply tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml]: Done!

File: tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml (hunk 3 above)
Comment (alyssa-sm) [reply tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml]: Done!

File: tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml (hunk 4 above)
Comment (alyssa-sm) [reply tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml]: Changed the deployment name to `zonal-policies`.

File: tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    source: modules/network/vpc
+
+  - id: homefs
+    source: modules/file-system/filestore
+    use: [network]
+    settings:
+      local_mount: /home
+
+  # Partition which permits a specific zone
+  - id: zonal_node_group
```
Comment (harshthakkar01) [inline tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml:44]: nodeset

Comment (harshthakkar01) [reply tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml:44]: It should be zonal-nodeset

File: tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      enable_placement: false
+
+  - id: zonal_partition
+    source: community/modules/compute/schedmd-slurm-gcp-v6-partition
+    use: [zonal_node_group]
+    settings:
+      partition_name: zonal
+
+  # Partition which allows a total of 3 zones
+  - id: multizonal_node_group
```
Comment (harshthakkar01) [inline tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml:60]: It should be multizonal-nodeset

**Comment by harshthakkar01** [comment]
you might want to rebase this pr.

File: tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml (hunk 5 above)
Comment (alyssa-sm) [reply tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml]: Done

File: tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml (hunk 6 above)
Comment (alyssa-sm) [reply tools/validate_configs/test_configs/zone-policies-slurm-v6.yaml]: Done!

**Pull request merged**